src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,52 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final double MIN_RANDOM_VALUE = 1.0;
+    public static final double MAX_RANDOM_VALUE = 20.0;
+    public static final int FIGURE_TYPES_COUNT = 5;
+    public static final int DEFAULT_RADIUS = 10;
+    private final ColorSupplier colorSupplier;
+    private final Random random = new Random();
+
+    public FigureSupplier(ColorSupplier colorSupplier) {
+        this.colorSupplier = colorSupplier;
+    }
+
+    public Figure getRandomFigure() {
+
+        int figureType = random.nextInt(FIGURE_TYPES_COUNT);
+        String color = colorSupplier.getRandomColor();
+        switch (figureType) {
+            case 0:
+                double squareSide = getRandomDoubleInRange(MIN_RANDOM_VALUE, MAX_RANDOM_VALUE);
+                return new Square(squareSide, color);
```

Comment:
No reason to write return in all cases, it is possible to define method level variable and populate it in cases, to avoid writing return